addons/stock_account/models/product.py

```diff
@@ -381,6 +381,11 @@ def _run_fifo_vacuum(self, company=None):
             }
             vacuum_svl = self.env['stock.valuation.layer'].sudo().create(vals)
 
+            # If some negative stock were fixed, we need to recompute the standard price.
+            product = self.with_context(force_company=company.id)
+            if product.cost_method == 'average' and not float_is_zero(product.quantity_svl, precision_rounding=self.uom_id.rounding):
+                product.sudo().write({'standard_price': product.value_svl / product.quantity_svl})
```

Comment:
It could be good to detail why this computation is correct while the standard price is computed differently here:

https://github.com/odoo/odoo/blob/7a02bdda12c161acb5b764866070b83872a7e566/addons/stock_account/models/stock_move.py#L307-L329

Reply:
we have some weird rules to update the avco when processing a receipt, like don't set it to 0 even if after the receipt you have 0 in stock. this is convenient to have an estimated avco when doing inventory adjustment or negative stock, but at the end the avco should represent the average of the value in your stock

here if the vacuum ran and some corrections entries added or removed some value in stock, the avco should be updated to represent what's really in stock, else value/qty is not the avco because of the quirks of product_price_update_before_done